dataland-framework-toolbox/src/main/kotlin/org/dataland/frameworktoolbox/frameworks/gdv/GdvFramework.kt

```diff
@@ -0,0 +1,548 @@
+import org.springframework.stereotype.Component
+import java.io.File
+
+/**
+ * Definition of the Heimathafen framework
+ */
+@Component
+class GdvFramework : InDevelopmentPavedRoadFramework(
+    identifier = "gdv",
+    label = "GDV/VÖB",
+    explanation = "Das GDV/VÖB Framework",
+    File("./dataland-framework-toolbox/inputs/gdv/dataDictionary-GDV-VOEB-GDV-VÖB ESG questionnaire.csv"),
+) {
+
+    @Suppress("LongMethod") // t0d0: fix detekt error later!
+    override fun customizeHighLevelIntermediateRepresentation(framework: Framework) {
+        val berichtsPflicht = framework.root
+            .getOrNull<ComponentGroup>("general")
+            ?.getOrNull<ComponentGroup>("masterData")
+            ?.getOrNull<YesNoComponent>("berichtsPflicht")
+        require(berichtsPflicht != null) {
+            "The field with the label \"berichtsPflicht\" must exist in the gdv framework."
+        }
+
+        val esgBerichte = framework.root
+            .getOrNull<ComponentGroup>("allgemein")
+            ?.getOrNull<ComponentGroup>("esgBerichte")
+        require(esgBerichte != null) {
+            "The component group with the label \"esgBerichte\" must exist in the gdv framework."
+        }
+
+        val nachhaltigkeitsberichte = esgBerichte.getOrNull<YesNoComponent>("nachhaltigkeitsberichte")
+        require(nachhaltigkeitsberichte != null) {
+            "The field with the label \"nachhaltigkeitsberichte\" must exist in the gdv framework."
+        }
+
+        val unGlobalConceptPrinzipien = framework.root
+            .getOrNull<ComponentGroup>("allgemein")
+            ?.getOrNull<ComponentGroup>("unGlobalConceptPrinzipien")
+        require(unGlobalConceptPrinzipien != null) {
+            "The section with the label \"unGlobalConceptPrinzipien\" must exist in the gdv framework."
+        }
+
+        val mechanismenZurUeberwachungDerEinhaltungDerUngcp =
+            unGlobalConceptPrinzipien.getOrNull<YesNoComponent>("mechanismenZurUeberwachungDerEinhaltungDerUngcp")
+        require(mechanismenZurUeberwachungDerEinhaltungDerUngcp != null) {
+            "The field with the label \"mechanismenZurUeberwachungDerEinhaltungDerUngcp\" " +
+                "must exist in the gdv framework."
+        }
+
+        val oecdLeitsaetze = framework.root
+            .getOrNull<ComponentGroup>("allgemein")
+            ?.getOrNull<ComponentGroup>("oecdLeitsaetze")
+        require(oecdLeitsaetze != null) {
+            "The section with the label \"oecdLeitsaetze\" must exist in the gdv framework."
+        }
+
+        val mechanismenZurUeberwachungDerEinhaltungDerOecdLeitsaetze =
+            oecdLeitsaetze.getOrNull<YesNoComponent>("mechanismenZurUeberwachungDerEinhaltungDerOecdLeitsaetze")
+        require(mechanismenZurUeberwachungDerEinhaltungDerOecdLeitsaetze != null) {
+            "The field with the label \"mechanismenZurUeberwachungDerEinhaltungDerOecdLeitsaetze\" " +
+                "must exist in the gdv framework."
```

Comment:
Why do we need all these the field must exist checks?

Reply:
I think because we either want to use the component-group-api to add components to these groups, or because we want to manually add a dependency to these groups for some other component.
In both cases it needs to be assured, that the group or component exists.

Reply:
Mostly removed during refactoring